core/src/main/scala/scalaz/syntax/ApplicativeBuilder.scala

```diff
@@ -162,3 +162,17 @@ private[scalaz] trait ApplicativeBuilder[M[_], A, B] {
   }
 
 }
+
+// outside of ApplicativeBuilder for bincompat
+abstract class ApplicativeBuilderSyntax {
```

Comment:
`ApplicativeBuilder` is sealed, right? Wouldn't it be safe to add these to it? I know that at a bytecode level it looks binary incompatible, but that's only if someone's used Java to extend `ApplicativeBuilder` already.

If not, could this go onto an `ApplicativeBuilder` companion, instead, so the enrichment shows up without importing `syntax.parallel._`?

Reply:
can't add it because of bincompat

Reply:
hmm, the first one could go on the companion, but I'm not sure if the 3...12 variants would see it. Technically it should be seen because it's a companion of an enclosing scope. Might be worth trying.

Reply:
@hrhino the answer is no, it can't see the syntax in the companion. I am compiling with 2.10 because that's necessary in 7.2, it may well be fixed in 2.12

:cry:

Reply:
noooo... this doesn't work either because `self.a` and `self.b` are in the outer scope and I don't think we can access it!

```scala
  implicit final class ApplicativeBuilderOps3[M[_], A, B, C](
    private val self: ApplicativeBuilder[M, A, B]#ApplicativeBuilder3[C]) {
    def parApply[Z](f: (A, B, C) => Z)(implicit ap: Apply[λ[α => M[α] @@ Parallel]]): M[Z] =
      Tag.unwrap(ap.apply3(
                   Tag[M[A], Parallel](self.a),
                   Tag[M[B], Parallel](self.b),
                   Tag[M[C], Parallel](self.c)
                 )(f))

    def parTupled(implicit ap: Apply[λ[α => M[α] @@ Parallel]]): M[(A, B, C)] =
      parApply(Tuple3.apply)
  }
```